docs/reference/concepts/02-provider.mdx

```diff
@@ -0,0 +1,184 @@
+
+- `openfeature-provider-<language name>`
+- `<vender/tool name>-openfeature-provider-<language name>`
+
+:::note
+
+It is **not** required that a provider is open source, but highly recommended.
+
+:::
+
+#### Artifact name
+
+The name of the artifact will vary based on the package manager. Ultimately, it's up to the publisher to decide what name makes the most sense. However, the following items should be used in the package name:
+
+- OpenFeature (can also be stylized as openfeature or open-feature)
+- Vender/tool name
+- The word `provider` should be included in the name
+
+An example of what it could look like in NPM: `@openfeature/flagd-provider`
+
+### Examples
+
+<Tabs groupId="code">
+<TabItem value="js" label="Typescript">
+
+```ts
+import { Provider, ResolutionDetails } from '@open';
+
+export class MyFeatureProvider implements Provider {
+  readonly metadata = {
+    name: 'My Feature Provider',
+  } as const;
+
+  resolveBooleanEvaluation(flagKey: string, defaultValue: boolean): Promise<ResolutionDetails<boolean>> {
+    // code to resolve boolean details
+  }
+
+  resolveStringEvaluation(flagKey: string, defaultValue: string): Promise<ResolutionDetails<string>> {
+    // code to resolve string details
+  }
+
+  resolveNumberEvaluation(flagKey: string, defaultValue: number): Promise<ResolutionDetails<number>> {
+    // code to resolve number details
+  }
+
+  resolveObjectEvaluation<T extends object>(flagKey: string, defaultValue: T): Promise<ResolutionDetails<T>> {
+    // code to resolve object details
+  }
+}
+```
+
+</TabItem>
+<TabItem value="java" label="Java">
+
+```java
+package dev.openfeature.javasdk;
+
+import lombok.Getter;
+
+public class MyFeatureProvider implements FeatureProvider {
+    @Getter
+    private final String name = "My Feature Provider";
+
+    @Override
+    public Metadata getMetadata() {
+        return new Metadata() {
+            @Override
+            public String getName() {
+                return name;
+            }
+        };
+    }
+
+    @Override
+    public ProviderEvaluation<Boolean> getBooleanEvaluation(String key, Boolean defaultValue, EvaluationContext ctx, FlagEvaluationOptions options) {
+      // code to resolve boolean details
+    }
+
+    @Override
+    public ProviderEvaluation<String> getStringEvaluation(String key, String defaultValue, EvaluationContext ctx, FlagEvaluationOptions options) {
+      // code to resolve string details
+    }
+
+    @Override
+    public ProviderEvaluation<Integer> getIntegerEvaluation(String key, Integer defaultValue, EvaluationContext ctx, FlagEvaluationOptions options) {
+      // code to resolve number details
+    }
+
+    @Override
+    public <T> ProviderEvaluation<T> getObjectEvaluation(String key, T defaultValue, EvaluationContext invocationContext, FlagEvaluationOptions options) {
+      // code to resolve object details
+    }
+}
+```
+
+</TabItem>
+<TabItem value="csharp" label="C#">
+
+```csharp
+using OpenFeature.SDK.Model;
```

Comment:
Just missing a using, but the code below is fine, there will be a compile error with no returns on the Resolve functions but that's fine I think.

```suggestion
using OpenFeature.SDK;
using OpenFeature.SDK.Model;
```